Adyen/HttpClient/HttpClientWrapper.cs

```diff
@@ -0,0 +1,140 @@
+        }
+
+        public string Request(string endpoint, string json, bool isApiKeyRequired, RequestOptions requestOptions = null)
+        {
+            return RequestAsync(endpoint, json, isApiKeyRequired, requestOptions).GetAwaiter().GetResult();
+        }
+
+        public async Task<string> RequestAsync(string endpoint, string json, bool isApiKeyRequired, RequestOptions requestOptions = null)
+        {
+            using (var request = GetHttpRequestMessage(endpoint, isApiKeyRequired, json, requestOptions))
+            using (var httpResponseMessage = await _httpClient.SendAsync(request))
+            {
+                var responseText = await httpResponseMessage.Content.ReadAsStringAsync();
+                httpResponseMessage.EnsureSuccessStatusCode();
+                return responseText;
+            }
+        }
+
+        public string Post(string endpoint, Dictionary<string, string> postParameters)
+        {
+            var dictToString = QueryString(postParameters);
+
+            var content = new StringContent(dictToString, _encoding, "application/x-www-form-urlencoded");
+            using (var request = new HttpRequestMessage(HttpMethod.Post, endpoint) {Content = content})
+            using (var response = _httpClient.SendAsync(request).GetAwaiter().GetResult())
+            {
+                response.EnsureSuccessStatusCode();
+                return response.Content.ReadAsStringAsync().GetAwaiter().GetResult();
+            }
+        }
+
+        public HttpRequestMessage GetHttpRequestMessage(string endpoint, bool isApiKeyRequired, string json, RequestOptions requestOptions)
+        {
+            var httpWebRequest = new HttpRequestMessage(HttpMethod.Post, endpoint)
+            {
+                Content = new StringContent(json, _encoding, "application/json")
+            };
+            httpWebRequest.Headers.Add("ContentType", "application/json");
+            httpWebRequest.Headers.Add("Accept-Charset", "UTF-8");
+            httpWebRequest.Headers.Add("Cache-Control", "no-cache");
+            httpWebRequest.Headers.Add("UserAgent", $"{_config.ApplicationName} {ClientConfig.UserAgentSuffix}{ClientConfig.LibVersion}");
+            if (!string.IsNullOrWhiteSpace(requestOptions?.IdempotencyKey))
+            {
+                httpWebRequest.Headers.Add("Idempotency-Key", requestOptions.IdempotencyKey);
+            }
+
+            //Use one of two authentication method.
+            if (isApiKeyRequired || _config.HasApiKey)
+            {
+                httpWebRequest.Headers.Add("x-api-key", _config.XApiKey);
+            }
+            else if (_config.HasPassword)
+            {
+                var authString = _config.Username + ":" + _config.Password;
+                var bytes = Encoding.UTF8.GetBytes(authString);
+                var credentials = Convert.ToBase64String(bytes);
+                httpWebRequest.Headers.Add("Authorization", "Basic " + credentials);
+            }
+
```

Comment:
I think here we can add the certificate validation for the local terminal API for POS 
similar to the [httpwebrequest](https://github.com/Adyen/adyen-dotnet-api-library/blob/develop/Adyen/HttpClient/HttpURLConnectionClient.cs#L200)
The validation is used to verify the certificate on the terminal against the public key.

Reply:
Actually, it's impossible to do it there. For `HttpClient` you can set up it through the `HttpClientHandler` during initialization only. It's impossible to configure it for a particular request. That's why I encapsulated this logic at `HttpClientExtensions.ConfigureHttpMessageHandler` method.

Reply:
Indeed but the `HttpClientExtensions.ConfigureHttpMessageHandler` is not used by the `HttpClient`. Could you provide us an example where to set up it through the `HttpClientHandler` during initialisation?